mail/spring-cloud-starter-stream-source-mail/src/test/java/org/springframework/cloud/stream/app/mail/source/MailSourceConfigurationTests.java

```diff
@@ -0,0 +1,207 @@
+		int n = 0;
+		while (n++ < 100 && (!MAIL_SERVER.isListening())) {
+			Thread.sleep(10);
+		}
+		assertTrue(n < 100);
+	}
+
+	@AfterClass
+	public static void cleanup() {
+		System.clearProperty("test.mail.server.port");
+		MAIL_SERVER.stop();
+	}
+
+	@IntegrationTest({ "protocol=imap","mail-url=imap://user:pw@localhost:${test.mail.server.port}/INBOX" })
+	public static class ImapPassTests extends MailSourceConfigurationTests {
+
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.imap(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertEquals("foo\r\n", received.getPayload());
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=imap","mail-url=imap://user:pw@localhost:${test.mail.server.port}/INBOX" })
+	public static class ImapFailTests extends MailSourceConfigurationTests {
+
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.imap(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertTrue(!received.getPayload().equals("Test Mail"));
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=pop3","mail-url=pop3://user:pw@localhost:${test.mail.server.port}/INBOX" })
+	public static class Pop3PassTests extends MailSourceConfigurationTests {
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.pop3(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertEquals("foo\r\n", received.getPayload());
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=pop3","mail-url=pop3://user:pw@localhost:${test.mail.server.port}/INBOX" })
+	public static class Pop3FailTests extends MailSourceConfigurationTests {
+
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.pop3(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertTrue(!received.getPayload().equals("Test Mail"));
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=imap", "idleImap=true","mail-url=imap://user:pw@localhost:${test.mail.server.port}/INBOX" })
```

Comment:
M-m-m. Does it really work for properties camel sty and dash delimiter? Cool! :smile: